include/revng/Model/ABI.h

```diff
@@ -6,24 +6,213 @@
 
 #include "llvm/ADT/StringRef.h"
 
+#include "revng/Model/Architecture.h"
 #include "revng/Model/TupleTree.h"
 #include "revng/Support/Debug.h"
 #include "revng/Support/YAMLTraits.h"
 
 namespace model::abi {
 
-enum Values { Invalid, SystemV_x86_64, Count };
+enum Values {
+  /// Unknown and/or unsupported ABI.
+  Invalid,
 
-inline llvm::StringRef getName(Values V) {
+  /// 64-bit SystemV ABI for x86 processor architecture.
+  /// The latest version of the documentation can be found
+  /// \sa https://gitlab.com/x86-psABIs/x86-64-ABI
+  SystemV_x86_64,
+
+  /// 32-bit SystemV ABI for x86 processor architecture.
+  /// The latest version of the documentation can be found
+  /// \sa https://gitlab.com/x86-psABIs/i386-ABI/-/tree/hjl/x86/master
+  SystemV_x86,
+
+  /// A gcc specific modification of the 32-bit SystemV ABI for x86 processor
+  /// architecture. It allows three first GPR-sized arguments to be passed
+  /// using the EAX, EDX, and ECX registers.
+  /// \sa `regparm` x86 function attribute.
+  SystemV_x86_regparm_3,
+
+  /// A gcc specific modification of the 32-bit SystemV ABI for x86 processor
+  /// architecture. It allows two first GPR-sized arguments to be passed
+  /// using the EAX, and ECX registers.
+  /// \sa `regparm` x86 function attribute.
+  SystemV_x86_regparm_2,
+
+  /// A gcc specific modification of the 32-bit SystemV ABI for x86 processor
+  /// architecture. It allows the first GPR-sized argument to be passed
+  /// using the EAX register.
+  /// \sa `regparm` x86 function attribute.
+  SystemV_x86_regparm_1,
+
+  /// 64-bit Microsoft ABI for x86 processor architecture.
+  /// The documentation can be found
+  /// \sa https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention
+  Microsoft_x64,
```

Comment:
What about the `PASCAL` calling convention? It's quite popular in win32 API.

Reply:
No-o-o, that's what I was afraid of. Mostly because it was so hard to find a reliable test case for it. BUT, from what I seem to be able to gather now, `PASCAL` is exactly same as `__stdcall` except the stuff is pushed to stack in the reverse order (which doesn't matter for current implementation - since it doesn't really handle stack yet).

Reply:
ACK, we want PASCAL too.

Reply:
I've added it.